src/main/resources/com/tresys/pcap/xsd/pcap.dfdl.xsd

```diff
@@ -203,20 +203,21 @@ The message root is 'PCAP'.
 
   <xs:element name="LinkLayer">
     <xs:complexType>
-      <xs:choice>
-        <xs:element ref="pcap:Ethernet" />
-      </xs:choice>
+      <xs:sequence>
+        <xs:element ref="pcap:Ethernet">
+          <xs:annotation>
+            <xs:appinfo source="http://www.ogf.org/dfdl/">
+            <dfdl:discriminator message="Link layer type is not Ethernet" test="{ xs:unsignedInt(../../../PCAPHeader/Network) eq 1 }" />
+            </xs:appinfo>
+          </xs:annotation>
+        </xs:element>
+      </xs:sequence>
```

Comment:
This was originally a choice to make it more clear where to add new link layer types, which I know there is some interest in doing. Can we make this a choice dispatch similar to what you have below?

Reply:
Will do. There's only one choice here, so I have to have some If-then logic to synthesize another possible value for the choice so that the 2nd alternative can be the error path. But that's ok.